### PR TITLE
Get user list

### PR DESCRIPTION
**Pull request by shernanm**
## Summary

* An endpoint is implemented that allows to list all the users that have registered and only authenticated users can access this data.

## Trello Card

https://trello.com/c/svlrGKj9

### CODE REVIEW

File: app/controllers/user.js [hunk 1]
```diff
@@ -26,3 +26,14 @@ exports.signInUser = (req, res, next) => {
     .then(token => res.status(200).send({ token }))
     .catch(next);
 };
+
+exports.getUserList = (req, res, next) => {
+  logger.info('getUserList method start.');
```
Comment (raparicio6) [inline app/controllers/user.js]: Log the req data for example the query

File: app/controllers/user.js [hunk 2]
```diff
@@ -26,3 +26,14 @@ exports.signInUser = (req, res, next) => {
...
+  const { limit, page } = req.query;
+  const offset = req.skip;
+  return userService
+    .findAllUser(limit, offset)
+    .then(foundUsers => utils.mapperUserList(foundUsers, limit, page))
```
Comment (raparicio6) [inline app/controllers/user.js]: Import mapper method with destructuring

File: app/controllers/user.js [hunk 3]
```diff
@@ -26,3 +26,14 @@ exports.signInUser = (req, res, next) => {
     .then(token => res.status(200).send({ token }))
     .catch(next);
 };
+
+exports.getUserList = (req, res, next) => {
+  logger.info('getUserList method start.');
+  const { limit, page } = req.query;
+  const offset = req.skip;
+  return userService
+    .findAllUser(limit, offset)
```
Comment (raparicio6) [inline app/controllers/user.js]: All users

File: app/controllers/user.js [hunk 4]
```diff
@@ -26,3 +26,14 @@ exports.signInUser = (req, res, next) => {
...
+    .then(foundUsers => utils.mapperUserList(foundUsers, limit, page))
+    .then(response => res.status(200).send(response))
```
Comment (raparicio6) [inline app/controllers/user.js]: You dont need to set the 200. It is set by default

File: app/services/users.js [hunk 5]
```diff
@@ -36,3 +35,13 @@ exports.signIn = (email, password) =>
...
     });
+
+exports.findAllUser = (limit, offset) =>
+  User.findAndCountAll({
+    limit,
+    offset,
+    attributes: ['id', 'firstName', 'lastName', 'email']
+  }).catch(err => {
+    logger.info('Error trying to find the users');
+    throw errors.databaseError(`${err}`);
```
Comment (raparicio6) [inline app/services/users.js]: If err is a string you dont need to put the ‘ and ${}

File: app/utils.js [hunk 6]
```diff
@@ -27,3 +27,14 @@ exports.validateToken = token =>
   jwt.verifyAsync(token, session.secret).catch(() => {
     throw errors.verifyTokenError();
   });
+
+exports.mapperUserList = (usersData, limit, page) => {
```
Comment (raparicio6) [inline app/utils.js]: Use destructuring when receiving usersData parameter

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: app/controllers/user.js [hunk 7]
```diff
@@ -1,13 +1,12 @@
-const utils = require('../utils');
-const userService = require('../services/users');
+const { passwordEncryption, mapperUserList } = require('../utils');
+const { userRegister, signIn, findAllUsers } = require('../services/users');
 const logger = require('../logger');
 
 exports.createUser = (req, res, next) => {
-  logger.info('createUser method start.');
+  logger.info(`CreateUser method start, request methods: ${req.method}, endpoint: ${req.path}.`);
```
Comment (raparicio6) [inline app/controllers/user.js]: always log body (if it is a post) and headers

File: app/middlewares/session.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const utils = require('../utils');
+const errors = require('../errors');
+
+exports.tokenValidator = (req, res, next) => {
+  const { token } = req.headers;
+  if (token) {
+    return utils
+      .validateToken(token)
+      .then(() => next())
+      .catch(error => {
```
Comment (raparicio6) [inline app/middlewares/session.js]: remove unnecesary { }

**Review by raparicio6 — CHANGES_REQUESTED** [review]

**Review by raparicio6 — APPROVED** [review]

**Review by molinajulian — APPROVED** [review]

**Pull request merged**